docs/csharp/quick-starts/introduction-to-classes.md

```diff
@@ -0,0 +1,177 @@
+
+```csharp
+this.Number = accountNumberSeed.ToString();
+accountNumberSeed++;
+```
+
+Type `dotnet run` to see the results.
+
+## Create deposits and withdrawals
+
+Your bank account class needs to accept deposits and withdrawals to work correctly. Let's implement deposits and withdrawals by creating a journal of every transaction for the account. That has a few advantages or simply updating the balance on each transaction. The history can be used to audit all transactions and manage daily balances. 
+
+Let's start by creating a new type to represent a transaction. This is a simple type that doesn't have any responsibilities. It needs a few properties. Create a new file named ***Transaction.cs***. Add the following code to it:
+
+[!code-csharp[Transaction](../../../samples/csharp/classes-quickstart/Transaction.cs "Transaction declaration")]
+
+Now, let's add a <xref:System.Collections.Generic.List%601> of `Transaction` objects to the `BankAccount` class. Add the following declaration:
+
+[!code-csharp[TransactionDecl](../../../samples/csharp/classes-quickstart/BankAccount.cs#TransactionDeclaration "Transaction declaration")]
+
+The <xref:System.Collections.Generic.List%601> class requires you to import a different namespace. Add the following at the beginning of **BankAccount.cs**:
+
+```csharp
+using System.Collections.Generic;
+```
+
+Now, let's change how the `Balance` is reported.  It can be found by summing the values of all transactions. Modify the declaration of `Balance` in the `BankAccount` class to the following:
+
+[!code-csharp[BalanceComputation](../../../samples/csharp/classes-quickstart/BankAccount.cs#BalanceComputation "Computing the balance")]
+
+This example shows an important aspect of ***properties***. You're now computing the balance when another programmer asks for the value. Your computation enumerates all transactions, and provides the sum as the current balance.
+
+Next implement the `MakeDeposit` and `MakeWithdrawal` methods:
+
+[!code-csharp[DepositAndWithdrawal](../../../samples/csharp/classes-quickstart/BankAccount.cs#DepositAndWithdrawal "Make deposits and withdrawals")]
+
+The constructor should get one change so that it adds an initial transaction, rather than updating the balance directly. Since you already wrote the `MakeDeposit` method, call it from your constructor. The finished constructor should look like this:
+
+[!code-csharp[Constructor](../../../samples/csharp/classes-quickstart/BankAccount.cs#Constructor "The final version of the constructor")]
+
+The <xref:System.DateTime.Now> is a property that returns the current date and time. Test this by adding a few deposits and withdrawals in your `Main` method:
+
+```csharp
+account.MakeWithdrawal(500, DateTime.Now, "Rent payment");
+Console.WriteLine(account.Balance);
+account.MakeDeposit(100, DateTime.Now, "friend paid me back");
+Console.WriteLine(account.Balance);
```

Comment:
There are two major issues here that really should be addressed:
1. You can create an account with a negative balance. 
2. You can overdraw your account, since there's no check to make sure that the balance is adequate to cover a withdrawal.

Reply:
Good points. I added those.